src/builder.rs

```diff
@@ -362,7 +362,20 @@ fn append_dir(dst: &mut Write, path: &Path, src_path: &Path, mode: HeaderMode) -
     append_fs(dst, path, &stat, &mut io::empty(), mode, None)
 }
 
-fn prepare_header(dst: &mut Write, header: &mut Header, path: &Path) -> io::Result<()> {
+fn prepare_header(size: u64, entry_type: u8) -> Header {
+    let mut header = Header::new_gnu();
+    header.as_gnu_mut().unwrap().name[..13].clone_from_slice(b"././@LongLink");
+    header.set_mode(0o644);
+    header.set_uid(0);
+    header.set_gid(0);
+    header.set_mtime(0);
+    header.set_size(size + 1);
```

Comment:
I think this is just the same as befeore, but I feel like a comment was missing from before as well. If you can recall why the `+1` happens here, could a comment be added indicating as to why?

Reply:
I don't know either, it was introduced by this commit https://github.com/alexcrichton/tar-rs/commit/bbbe5b129099e214fb74aba4957d74fd97c6846c : "Apparently GNU does this, so we should too!". That's [true](https://git.savannah.gnu.org/cgit/tar.git/tree/src/create.c#n541), but I don't know why exactly, maybe to have a null-terminated name ?
I added a comment.